webrender/src/border.rs

```diff
@@ -1233,31 +952,51 @@ fn add_corner_segment(
         return;
     }
 
-    border_segments.push(BorderSegmentInfo {
-        task_rect,
-        segment,
-        radius,
-        widths,
-    });
+    if image_rect.size.width <= 0. || image_rect.size.height <= 0. {
+        return;
+    }
```

Comment:
Just double-checking, it's fine to avoid pushing the border segment here since we request each segment to the render task cache separately, right? That's much nicer :)

The reason the previous code couldn't do that was because we'd just cache all the instances for all the segments at the same time, which meant IIRC that if any of the segments was skipped, another border with an actual border size but the same cache key could reuse the wrong instances...

Reply:
Yep, that's right - the brush / batch code didn't previously support a render task / texture per-segment, but we can take advantage of that now :)